client/invoke.go

```diff
@@ -0,0 +1,109 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"os"
+
+	"github.com/fnproject/fn_go/provider"
+	"github.com/go-openapi/runtime/logger"
+)
+
+func Invoke(provider provider.Provider, appName, fnID string, content io.Reader, output io.Writer, method string, env []string, contentType string, includeCallID bool) error {
+
+	u, err := provider.CallURL(appName)
+	if err != nil {
+		return err
+	}
+	u.Path = "invoke/" + fnID
```

Comment:
I see that there are complications with using the swagger generated client due to provider-specifics, etc.

Could we, however, still support invoke in fn_go and just call that in the CLI?

Reply:
we could move these things that look like client extensions down into fn_go i guess, but I wonder if there was a reason that wasn't done before though...

Reply:
`/r/` wasn't ever intended to be a first class part of the API (though in practice it kinda was), since users probably want custom routes w/ e.g. dns they'd configure. but in any case, invoke a function now seems like it is part of the api proper

Reply:
as such, we may want to version it instead of `/invoke`... we should maybe clean those up, there are a few endpoints down there